### PR TITLE
Add ConnectionTuple to replace ByteKey

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Adds an embedded `network.ConnectionTuple` type that is comparable. 

### Motivation

1. This can be used in place of the `ByteKey` function, which created a byte slice.
2. This lays the groundwork to fix a race condition with how failed TCP connections are handled (coming in a future PR soon).

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Comment by guyarb** [comment]
> This can be used in place of the ByteKey function, which created a byte slice.

Does the change have any impact on performance?

> This lays the groundwork to fix a race condition with how failed TCP connections are handled (coming in a future PR soon).

Can you elaborate more how it solves a race condition?

File: pkg/network/tracer/connection/ebpf_tracer.go [hunk 1]
```diff
@@ -314,12 +314,15 @@ func (t *ebpfTracer) Start(callback func(*network.ConnectionStats)) (err error)
 		return fmt.Errorf("error initializing port binding maps: %s", err)
 	}
 
+	t.closeConsumer.Start(callback)
+	t.failedConnConsumer.Start()
+
 	if err := t.m.Start(); err != nil {
+		t.closeConsumer.Stop()
+		t.failedConnConsumer.Stop()
```
Comment (guyarb) [inline pkg/network/tracer/connection/ebpf_tracer.go:322]: worth adding inline documentation here, someone might change it in the future without understanding why you've made this change

**Review by guyarb — APPROVED** [review]
LGTM for USM

**Review by derekwbrown — APPROVED** [review]
approved for (2) wkit owned files

**Pull request merged**